src/backend/ErrorData.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright 2018 The NXT Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "backend/ErrorData.h"
+
+namespace backend {
+
+    ErrorData::ErrorData() = default;
+
+    ErrorData::ErrorData(std::string message) : mMessage(std::move(message)) {
+    }
+
+    void ErrorData::AppendBacktrace(const char* file, const char* function, int line) {
+        BacktraceRecord record;
+        record.file = file;
+        record.function = function;
+        record.line = line;
+
+        mBacktrace.push_back(record);
```

Comment:
nit: I would std::move(record) in case you change the members to more complex types later.

Reply:
Done